src/main/java/prefs/Preferences.java

```diff
@@ -18,13 +17,8 @@
 
     public GlobalConfig global;
 
-    public Preferences(boolean isTestMode) {
-        if (isTestMode) {
-            this.fileHandler = new ConfigFileHandler(DIRECTORY, TEST_CONFIG_FILE);
-        } else {
-            this.fileHandler = new ConfigFileHandler(DIRECTORY, GLOBAL_CONFIG_FILE);
-        }
-
+    public Preferences(Optional<String> configFileName) {
```

Comment:
I prefer the original since the parameter can only be one or the other, but if we have a function that initialises this for both (as mentioned on my other comment), this should be fine.